conductor/lib/downloader.py

```diff
@@ -1270,23 +1270,32 @@ def make_url(path):
         '''
         TODO: get rid of this hardcoding!!!
```

Comment:
This **make_url** method feels like it's trying to figure out some core routing logic that I would think is applicable to a broader range of code, than just the downloader?  For instance, the uploader will need to use this same logic, yes?  Perhaps encapsulating this logic in api_client.py (or common.py) would be better?